lib/heuristics/scheduling_heuristic.rb

```diff
@@ -1086,22 +1062,23 @@ def prepare_output_and_collect_routes(vrp)
     def select_point(insertion_costs, empty_route)
       ### chose the most interesting point to insert according to [insertion_costs] ###
       if empty_route
-        if @routes_referents.empty?
+        if @candidate_routes.all?{ |_vehicle_id, all_routes| all_routes.all?{ |_day, route_data| route_data[:stops].empty? } }
           # chose closest with highest frequency
-          to_consider_set = insertion_costs.group_by{ |s| @services_data[s[:id]][:visits_number] }.sort_by{ |nb, _set| nb}.reverse.first[1]
+          to_consider_set = insertion_costs.group_by{ |s| @services_data[s[:id]][:raw].visits_number }.sort_by{ |nb, _set| nb }.reverse.first[1]
```

Comment:
we can order correctly instead of `reverse` it
```suggestion
          to_consider_set = insertion_costs.group_by{ |s| @services_data[s[:id]][:raw].visits_number }.sort_by{ |nb, _set| -nb }.first[1]
```